src/analyzer/vamp/vamppluginloader.cpp

```diff
@@ -1,58 +1,137 @@
-/*
- * vamppluginloader.cpp
- *
- *  Created on: 23/jan/2012
- *      Author: Tobias Rafreider
- *
- * This is a thread-safe wrapper class around Vamp's
- * PluginLoader class.
- */
 #include "analyzer/vamp/vamppluginloader.h"
 
-using Vamp::Plugin;
-using Vamp::PluginHostAdapter;
+#include <mutex>
 
-VampPluginLoader* VampPluginLoader::s_instance = NULL;
-QMutex VampPluginLoader::s_mutex;
+#include <QCoreApplication>
+#include <QDesktopServices>
+#include <QDir>
+#include <QStringList>
+#include <QStringBuilder>
 
-VampPluginLoader::VampPluginLoader() {
-    m_pVampPluginLoader = Vamp::HostExt::PluginLoader::getInstance();
+#include "util/logger.h"
+
+#ifdef __WINDOWS__
+    #include <windows.h>
+    #define PATH_SEPARATOR ";"
+#else
+    #define PATH_SEPARATOR ":"
+#endif
+
+
+namespace mixxx {
+
+namespace {
+
+Logger kLogger("VampPluginLoader");
+
+Vamp::HostExt::PluginLoader* pPluginLoader = nullptr;
+
+std::once_flag initPluginLoaderOnceFlag;
+
+// Initialize the VAMP_PATH environment variable to point to the default
+// places that Mixxx VAMP plugins are deployed on installation. If a
+// VAMP_PATH environment variable is already set by the user, then this
+// method appends to that.
+void initPluginPaths() {
+    const QLatin1String pathEnv(getenv("VAMP_PATH"));
```

Comment:
are we sure this is a latin1 string on all targets? I consider this to be a local 8 bit